src/log.cpp

```diff
@@ -310,16 +310,30 @@ void Logger::logToOutputs(LogLevel lev, const std::string &combined,
 //// *LogOutput methods
 ////
 
-void FileLogOutput::open(const std::string &filename)
+void FileLogOutput::setFile(const std::string &filename, u64 file_size_max)
 {
-	m_stream.open(filename.c_str(), std::ios::app | std::ios::ate);
+	actionstream << "Log messages are saved to " << filename << std::endl;
+
+	std::ifstream ifile(filename, std::ios::binary | std::ios::ate);
+	bool is_too_large = ifile.tellg() > static_cast<std::streamoff>(file_size_max);
```

Comment:
I'd also like this mechanism to be optional by setting `debug_log_size_max = 0` (or -1)